Data race and unsafe mutation of global `http.DefaultClient` in backup storage validation

### What is the issue?

The codebase unsafely mutates the global `http.DefaultClient` by modifying its `Timeout` and `Transport` fields during backup storage validation. This causes a data race and modifies the behavior of all other HTTP requests occurring concurrently across the entire OpenEverest server process.

In `internal/server/handlers/validation/backup_storage.go` (around line 225):
```go
c := http.DefaultClient
c.Timeout = timeoutS3AccessSec * time.Second
c.Transport = &http.Transport{
    TLSClientConfig: &tls.Config{
        InsecureSkipVerify: !verifyTLS,
    },
}
```

### Steps to Reproduce


### Steps to Reproduce
```markdown
1. Trigger a validation request for Backup Storage that executes the S3 validation logic in `backup_storage.go`.
2. Notice that the global `http.DefaultClient` is permanently mutated with a new `Timeout` and `Transport` for the lifetime of the process.
3. Concurrently execute any other server feature that relies on `http.DefaultClient`; it will inherit these hijacked settings, resulting in a data race.
```

### Relevant Logs / Output
```
==================
WARNING: DATA RACE
Write at 0x000104fea1a8 by goroutine 49:
  github.com/percona/everest/internal/server/handlers/validation.s3Access()
      /Users/lalit/openeverest/internal/server/handlers/validation/backup_storage.go:226 +0xa0
...
Goroutine 48 (running) created at:
  github.com/percona/everest/internal/server/handlers/validation.TestS3AccessRace()
      /Users/lalit/openeverest/internal/server/handlers/validation/backup_storage_race_test.go:16 +0x2a0
==================
--- FAIL: TestS3AccessRace (0.38s)
    testing.go:1712: race detected during execution of test

```

### OpenEverest Version

main branch

### Installation Method

Compiled from source / run via local backend setup

### Kubernetes Distribution

k3d (local test cluster)

### Environment Details

- **OS**: macOS / Linux local development environment
- **K8s Version**: v1.28
- **Provider**: Local (k3d)
- **Go Version**: 1.22+


### Possible Solution

Create a new `http.Client` instead of using the global default:

```go
c := &http.Client{
    Timeout: timeoutS3AccessSec * time.Second,
    Transport: &http.Transport{
        TLSClientConfig: &tls.Config{
            InsecureSkipVerify: !verifyTLS,
        },
    },
}
```

### Additional Context

N/A

--- a/internal/server/handlers/validation/backup_storage.go
+++ b/internal/server/handlers/validation/backup_storage.go
@@ -1,3 +1,17 @@
+// Copyright (C) 2026 The OpenEverest Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package validation
 
 import (
@@ -208,10 +222,11 @@ func s3Access(
 		endpoint = nil
 	}
 
-	c := http.DefaultClient
-	c.Timeout = timeoutS3AccessSec * time.Second
-	c.Transport = &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: !verifyTLS}, //nolint:gosec
+	c := &http.Client{
+		Timeout: timeoutS3AccessSec * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: !verifyTLS}, //nolint:gosec
+		},
 	}
 	// Create a new session with the provided credentials
 	sess, err := session.NewSession(&aws.Config{

